lib/dry/configurable/settings.rb

```diff
@@ -23,6 +23,35 @@ def initialize(elements = EMPTY_ARRAY)
         initialize_elements(elements)
       end
 
+      def replace(settings)
+        unless settings.is_a? Dry::Configurable::Settings
+          raise ArgumentError, "settings must be a Dry::Configurable::Settings"
+        end
+
+        settings.each do |setting|
+          self << setting
+        end
+      end
+
+      def merge(settings)
```

Comment:
Before we commit to this naming, I think it'd be worth contemplating whether we want a `#merge` that returns a new copy of the `Settings`, and a `merge!` that modifies in place (which this method as implemented is currently doing).

What do you think, @solnic? If we're going to follow that idea of yours and make the `Settings` more directly accessible via a `.settings` class method (instead of `._settings`), then having a slightly more expansive API on those objects could be handy.

(`#replace` OTOH doesn't need the bang on the end of its name because its only function is to mutate the receiver - we're mirroring the Ruby stdlib in this regard)

Reply:
I'm unsure of the difference between a merge! and replace, so I removed replaced in favor of merge, and merge!

Reply:
I think we had a bit of confusion just then. I wasn't advocating for us to rename `#replace` – that's a good name and we should keep it as it was before. Would you mind putting that back?

What I was actually musing on was about the behaviour of the `Settings#merge` method, whether it should mutate the `Settings` instance or return a _new copy_ of the the instance with the merged settings, leaving the original one alone. I think it might make sense of us to have both. But for the time being, let's rename your original `#merge` to `#merge!` for clarity.

Reply:
I can add replace back no problem, but the PR fix I made also already adds a non destructive "merge" and a destructive "merge!", like you said.

I'll go ahead and add replace back.